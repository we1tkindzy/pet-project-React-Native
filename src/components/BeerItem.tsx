import {useContext} from 'react';
import {View, Text, StyleSheet, Image, TouchableOpacity} from 'react-native';

import ThemeContext from '../context/AppContext';

interface Beer {
  id: number;
  name: string;
  tagline: string;
  image_url: string;
  brewers_tips: string;
}

interface BeerItemProps {
  beer: Beer;
  navigation: any;
}

const BeerItem = ({beer, navigation}: BeerItemProps) => {
  const {theme} = useContext(ThemeContext);

  const colorStyles = StyleSheet.create({
    textColor: {
      color: theme === 'dark' ? '#ffffff' : '#371005',
    },
    rightSectionBgColor: {
      backgroundColor: theme === 'dark' ? '#D9B597' : '#B6846B',
    },
  });

  return (
    <TouchableOpacity
      onPress={() => navigation.navigate('Beer', {beer})}
      style={styles.beer}>
      <View style={styles.leftSection}>
        <Image
          style={styles.beerImg}
          alt={beer.name}
          height={120}
          width={120}
          resizeMode="contain"
          source={{uri: `${beer.image_url}`}}
        />

        <Text style={styles.beerTagline}>{beer.tagline}</Text>
      </View>

      <View style={[styles.rightSection, colorStyles.rightSectionBgColor]}>
        <Text style={styles.beerTitle}>{beer.name}</Text>

        <Text style={styles.beerTips}>{beer.brewers_tips}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default BeerItem;

const styles = StyleSheet.create({
  beer: {
    width: '100%',
    marginBottom: 20,
    display: 'flex',
    flexDirection: 'row',
  },
  leftSection: {
    backgroundColor: '#B56735',
    borderBottomLeftRadius: 10,
    borderTopLeftRadius: 10,
    padding: 8,
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '40%',
  },
  beerImg: {
    margin: 'auto',
    marginBottom: 10,
  },
  beerTagline: {
    textAlign: 'center',
  },
  rightSection: {
    backgroundColor: '#B6846B',
    borderBottomRightRadius: 10,
    borderTopRightRadius: 10,
    padding: 10,
    width: '60%',
  },
  beerTitle: {
    color: '#371005',
    fontSize: 24,
    marginBottom: 10,
  },
  beerTips: {
    color: '#FFFFFF',
    fontSize: 14,
  },
});
