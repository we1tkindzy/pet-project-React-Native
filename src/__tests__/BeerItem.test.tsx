import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import BeerItem from '../components/BeerItem';

const testBeerData = {
  id: 1,
  name: 'Beer Name',
  tagline: 'tagline',
  image_url: 'img/beer.png',
  brewers_tips: 'tips',
};

it('should renders BeerItem component with data', () => {
  const beerItem = renderer.create(
    <BeerItem beer={testBeerData} navigation={jest.fn()} />,
  );
  expect(beerItem).toMatchSnapshot();
});
