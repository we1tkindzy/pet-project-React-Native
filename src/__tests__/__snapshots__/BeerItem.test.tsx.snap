// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders BeerItem component with data 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 20,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#B56735",
        "borderBottomLeftRadius": 10,
        "borderTopLeftRadius": 10,
        "display": "flex",
        "justifyContent": "space-between",
        "padding": 8,
        "width": "40%",
      }
    }
  >
    <Image
      alt="Beer Name"
      height={120}
      resizeMode="contain"
      source={
        {
          "uri": "img/beer.png",
        }
      }
      style={
        {
          "margin": "auto",
          "marginBottom": 10,
        }
      }
      width={120}
    />
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      tagline
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#B6846B",
          "borderBottomRightRadius": 10,
          "borderTopRightRadius": 10,
          "padding": 10,
          "width": "60%",
        },
        {
          "backgroundColor": "#B6846B",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#371005",
          "fontSize": 24,
          "marginBottom": 10,
        }
      }
    >
      Beer Name
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 14,
        }
      }
    >
      tips
    </Text>
  </View>
</View>
`;
